scripts/pyproject.toml

```diff
@@ -13,10 +13,10 @@ pygeos = "^0.12.0"
 pandas = "^1.4.2"
 click = "^8.1.2"
 
-[tool.poetry.dev-dependencies]
-ipykernel = "^6.13.0"
-isort = "^5.10.1"
```

Comment:
ipykernel is fine to remove; I tend to use that myself for interactive debugging of scripts. I'd have a preference for keeping isort though, because then (along with black) we have a deterministic formatting of any scripts, no matter who's editing it

Reply:
I added `isort` back in the latest commit.

I think it is nice to work on a repo where the dependencies are used (ideally in a CI job).  So for example, if we were running `black --check` in a CI job, that would justify the `black` dependency.  And we could have a job that exercised `isort` as well.  (FWIW, I think doing both of these things would be great - I love not thinking about whitespace or formatting and having tools take care of that for me.)

Reply:
Would you rather have those lint steps in a CI job or in the pre commit config? Or should we remove the pre commit config altogether?

Reply:
CI job

Reply:
This is probably controversial, but I find pre-commit hooks developer hostile.  I like to use commits as a place to persist work in progress.  Often that work in progress won't pass all tests (linter rules, etc).  Having a pre-commit hook that makes it harder to push work in progress commits feels like a hassle.  I think there is a ton of value in these rules, tests, etc.  CI jobs that prevent merging feels like the appropriate place to enforce them.

Reply:
Yes `--no-verify` is your friend ;) (`-n` for short)

While I often use --no-verify for exactly the reason you describe above, I also love pre-commit because it makes me to have to think so much less about linting in general.

Also, you don't _have_ to install the pre-commit hooks in a repo. So even if we have `.pre-commit-config.yaml` and use that also in CI to run the linting checks, you are not obliged to use that locally.

Reply:
Thanks, I do know how to disable/ignore them - just saying that I think in general a pre-commit hook makes it harder to push WIP commits.  I haven't done anything in this branch related to the pre-commit hooks, but good discussion.